### PR TITLE
 TM1637 functionality

### PR DESCRIPTION
**Pull request by Mynasru**
## Description: 
TM1637 functionality

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/tm1637/tm1637.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    void showDots(uint8_t dots, uint8_t* digits);
+    void showNumberBaseEx(int8_t base, uint16_t num, uint8_t dots = 0, bool leading_zero = false, uint8_t length = 4, uint8_t pos = 0);
+    
+    GPIOPin *dio_pin_;
+    GPIOPin *clk_pin_;
+
+  private:
+    uint8_t m_brightness;
+    unsigned int m_bitDelay;
```
Comment (OttoWinter) [inline esphome/components/tm1637/tm1637.h]:
```suggestion
    unsigned int bit_delay_; 
```

File: esphome/components/tm1637/tm1637.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    bool writeByte(uint8_t b);
+
+    void showDots(uint8_t dots, uint8_t* digits);
+    void showNumberBaseEx(int8_t base, uint16_t num, uint8_t dots = 0, bool leading_zero = false, uint8_t length = 4, uint8_t pos = 0);
+    
+    GPIOPin *dio_pin_;
+    GPIOPin *clk_pin_;
+
+  private:
+    uint8_t m_brightness;
```
Comment (OttoWinter) [inline esphome/components/tm1637/tm1637.h]:
```suggestion
    uint8_t brightness_;
```

File: esphome/components/tm1637/tm1637.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    void clear();
+    void showNumberDec(int num, bool leading_zero = false, uint8_t length = 4, uint8_t pos = 0);
+    void showNumberDecEx(int num, uint8_t dots = 0, bool leading_zero = false, uint8_t length = 4, uint8_t pos = 0);
+    void showNumberHexEx(uint16_t num, uint8_t dots = 0, bool leading_zero = false, uint8_t length = 4, uint8_t pos = 0);
+    uint8_t encodeDigit(uint8_t digit);
+
+  protected:
+    void setup_pins_();
+
+    void bitDelay();
```
Comment (OttoWinter) [inline esphome/components/tm1637/tm1637.h]:
```suggestion
    void bit_delay_();
```

**Review by OttoWinter — COMMENTED** [review]
Thanks!

- ESPHome uses a different naming/formatting convention. See contributing page - and if you un-mark this PR as a draft you can also see travis output which shows what needs to be changed.
- ESPHome's displays use a different approach for displays (see max7219 as an example). The "writer" lambda is given methods (like `print`) that print to an internal buffer. After that's done, the changes are applied. Please use that same approach here (you can probably copy the print etc functions from max7219)

**Comment by glmnet** [comment]
Closing because implemented in https://github.com/esphome/esphome/pull/946

**Pull request closed without merging**